astroquery/utils/tap/xmlparser/utils.py

```diff
@@ -24,11 +24,18 @@ def util_create_string_from_buffer(buffer):
     return ''.join(map(str, buffer))
 
 
-def read_http_response(response, outputFormat, correct_units=True):
-    astropyFormat = get_suitable_astropy_format(outputFormat)
+def read_http_response(response, output_format, correct_units=True):
+    astropy_format = get_suitable_astropy_format(output_format)
+
     # If we want to use astropy.table, we have to read the data
     data = io.BytesIO(response.read())
-    result = APTable.read(data, format=astropyFormat)
+
+    try:
+        result = APTable.read(io.BytesIO(gzip.decompress(data.read())), format=astropy_format)
+    except gzip.BadGzipFile:
```

Comment:
We still support python3.7, and `BadGzipFile` has only been added into 3.8, so a workaround is needed here. I see two ways, but maybe there are others (I don't have a preference of which one you do):

1) - except on `OSError` as `BadGzipFile` inherits for that one. Based on the docs, `EOFError` or `zlib.error` could be options, too.

2) Add a custom exception based on the python version. Somewhat ugly, but it is the closest to what you're checking right now, and would allow a trivial cleanup once 3.7 support is dropped.
```
try:
    # We need python > 3.7 for this
    from gzip import BadGzipFile
except ImportError:
    class BadGzipFile(OSError):
        pass
```

Reply:
Thank you very much. Working on this right now

Reply:
change applied. thank you again for the suggestion. very useful